pytorch_lightning/core/lightning.py

```diff
@@ -379,7 +387,18 @@ def all_gather(self, tensor: Union[torch.Tensor], group: Optional[Any] = None, s
         Return:
             A tensor of shape (world_size, batch, ...)
```

Comment:
```suggestion
            A tensor of shape (world_size, batch, ...), or if the input was a collection the output will also be a collection with tensors of this shape.
```

Reply:
the rest of the docs above need to be updated with the renaming you did. 
in fact, sphinx should warn about this and I'm not sure why docs build is passing on this.